README.md

```diff
@@ -0,0 +1,767 @@
+
+``` mermaid
+%%{ init: { 'flowchart': { 'curve': 'linear' } } }%%
+
+flowchart BT
+  *
+
+  msg/* --> *
+  subgraph msgGraph [ ]
+    msg/send --> msg/*
+    msg/receive --> msg/*
+  end
+
+  crud/* --> *
+  subgraph crudGraph [ ]
+    crud/read --> crud/*
+    crud/mutate --> crud/*
+
+    subgraph mutationGraph [ ]
+        crud/create --> crud/mutate
+        crud/update --> crud/mutate
+        crud/destroy --> crud/mutate
+    end
+  end
+
+  ... --> *
+```
+
+Conceptually it has this shape:
+
+``` ts
+{
+  "can": "*",
+  "args": {
+    "can": string, // Some other command
+    "args": {[string]: any} // That commad's arguments
+  },
+  // ...
+}
+```
+
+Since the nesting is fully redundant and infinitely nestable, it is instead used only in proof chains, and SHOULD NOT be invoked directly.
+
+## Arguments
+
+The `args` field MAY contain partially applied Arguments for the shape of data specified by the [Command]. For example, below is an example that constrains sending email from a particular address:
+
+``` js
+// Delegation
+{
+  "can": "msg/send",
+  "args": {
+    "from": "alice@example.com",
+    "title": "Coffee"
+  },
+  // ...
+}
+
+// Valid invocation
+{
+  "do": "msg/send",
+  "args": {
+    "from": "alice@example.com", // Matches above
+    "to": ["bob@example.com", "carol@elsewhere.example.com"],
+    "title": "Coffee",
+    "body": "Still on for coffee" // Matches above
+  },
+  // ...
+}
+```
+
+Any arguments MUST be taken verbatim and MUST NOT be further adjusted. For more flexible validation of Arguments, use [Conditions].
+
+Note that this also applies to arrays and objects. For example, the `to` array in this example is considered to be exact, so the Invocation fails validation in this case:
+
+``` js
+// Delegation
+{
+  "can": "msg/send",
+  "args": {
+    "from": "alice@example.com",
+    "to": ["bob@example.com"],
+  },
+  // ...
+}
+
+// INVALID Invocation
+{
+  "do": "msg/send",
+  "args": {
+    "from": "alice@example.com", // Matches above
+    "to": ["bob@example.com", "carol@elsewhere.example.com"], // Does not match exactly
+    "title": "Coffee",
+    "body": "Still on for coffee"
+  },
+  // ...
+}
+```
+
+The intended logic is expressible with [Conditions].
+
+## Conditions
+
+The `cond` field MUST contain any additional conditions. This concept is sometimes called a "caveat". Conditions constrain the capability in two ways:
+
+- Syntactic constraints on [Arguments] (length, regex, inclusion)
```

Comment:
Reading the preview, and it looks like `[Arguments]` may need a link definition.